ui/apps/platform/src/Containers/NetworkGraph/components/NetworkBreadcrumbs.tsx

```diff
@@ -0,0 +1,149 @@
+    const [isNamespaceOpen, setIsNamespaceOpen] = React.useState(false);
+
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    const {
+        cluster: selectedClusterName,
+        namespaces: selectedNamespaces,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        deployments: deploymentsFromUrl,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        remainingQuery,
+    } = getScopeHierarchy(searchFilter);
+    const selectedClusterId =
+        clusters.find((cluster) => cluster.name === selectedClusterName)?.id || '';
+    const { loading, error, namespaces } = useFetchClusterNamespaces(selectedClusterId);
+
+    const onClusterToggle = (newClusterOpenValue) => setIsClusterOpen(newClusterOpenValue);
+
+    const onNamespaceToggle = (newNamespaceOpenValue: boolean) =>
+        setIsNamespaceOpen(newNamespaceOpenValue);
+
+    const onClusterSelect = (_, value) => {
+        setIsClusterOpen((prevIsOpen: boolean) => !prevIsOpen);
+
+        if (value !== selectedClusterName) {
+            const modifiedSearchObject = { ...searchFilter };
+            modifiedSearchObject.Cluster = value;
+            setSearchFilter(modifiedSearchObject);
+        }
+    };
+
+    const onFilterNamespaces = useCallback(
+        (e: ChangeEvent<HTMLInputElement> | null, filterValue: string) =>
+            filterElementsWithValueProp(
+                filterValue,
+                namespaces.map((filteredNS) => (
+                    <SelectOption key={filteredNS} value={filteredNS}>
+                        <span>
+                            <NamespaceIcon /> {filteredNS}
+                        </span>
+                    </SelectOption>
+                ))
+            ),
+        [namespaces]
+    );
+
+    const onNamespaceSelect = (_, selected) => {
+        setIsNamespaceOpen((prevIsOpen: boolean) => !prevIsOpen);
+
+        const newSelection = selectedNamespaces.find((nsFilter) => nsFilter === selected)
+            ? selectedNamespaces.filter((nsFilter) => nsFilter !== selected)
+            : selectedNamespaces.concat(selected);
+
+        const modifiedSearchObject = { ...searchFilter };
+        modifiedSearchObject.Namespace = newSelection;
+        setSearchFilter(modifiedSearchObject);
+    };
+
+    const clusterSelectOptions: JSX.Element[] = clusters.map((cluster) => (
+        <SelectOption key={cluster.id} value={cluster.name}>
+            <span>
+                <ClusterIcon /> {cluster.name}
+            </span>
+        </SelectOption>
+    ));
+    const namespaceSelectOptions: JSX.Element[] = namespaces.map((namespace) => (
+        <SelectOption key={namespace} value={namespace}>
+            <span>
+                <NamespaceIcon /> {namespace}
+            </span>
+        </SelectOption>
+    ));
+
+    return (
+        <>
+            <Breadcrumb>
+                <BreadcrumbItem isDropdown>
```

Comment:
Do you think it would be worthwhile to encapsulate the cluster-related breadcrumb logic and namespaces-related breadcrumb logic into their own separate components?